### ISSUE TITLE
gpt-oss

### ISSUE THREAD

**Issue by ddilbazTT** [tenstorrent/tt-torch#1168]
Bring up `openai/gpt-oss-20b` model available on hugging face: https://huggingface.co/openai/gpt-oss-20b 
Start with single layer, then increase layers and run on multichip.

**Comment by ddilbazTT** [comment]
Need to install torch, torch-xla latest nightly releases:
```
pip install --pre torch torchvision --index-url https://download.pytorch.org/whl/nightly/cpu
pip install https://storage.googleapis.com/pytorch-xla-releases/wheels/tpuvm/torch_xla-2.8.0.dev-cp310-cp310-linux_x86_64.whl
```

**Comment by ddilbazTT** [comment]
I got the ttir to ttnn conversion working. I need to implement slice + reshape to achieve the following:

`index_tensor = scatter_indices.select(index_vector_dim, scatter_dims_to_operand_dims.index(dim))`

In order for slice and reshape, I needed to add layout attributes to the types.

**Comment by ddilbazTT** [comment]
I added scatter workaround. Later, I received the following error: 
```
2025-08-18 16:39:02.740 | critical |          Always | Can only work with bfloat16/float32 or int32/uint32 tensors (assert.hpp:107)
```

This was coming from reshape op that received ui8 input. The reason for this is because the following torch graph: 
```
def test_booleanAdd():
    class Basic(nn.Module):
        def __init__(self):
            super().__init__()

        def forward(self, x, y):
            return x & y
    x = torch.tensor([True, False, True])
    y = torch.tensor([False, True, True])
    tt_mod = torch.compile(Basic(), backend="tt-experimental")
    out = tt_mod(x, y)
```

becomes

```
module @SyncTensorsGraph.8 attributes {mhlo.cross_program_prefetches = [], mhlo.input_output_alias = [], mhlo.is_dynamic = false, mhlo.use_auto_spmd_partitioning = false} {
  sdy.mesh @mesh = <["x"=1, "y"=1]>
  func.func @main(%arg0: tensor<3xi1> {ttcore.shard_status = #ttcore.shard_status<unsharded>}, %arg1: tensor<3xi1> {ttcore.shard_status = #ttcore.shard_status<unsharded>}) -> (tensor<3xi1> {ttcore.shard_status = #ttcore.shard_status<unsharded>}) {
    %c = stablehlo.constant dense<0> : tensor<3xui8>
    %0 = stablehlo.convert %arg1 : (tensor<3xi1>) -> tensor<3xui8>
    %1 = stablehlo.convert %arg0 : (tensor<3xi1>) -> tensor<3xui8>
    %2 = stablehlo.and %0, %1 : tensor<3xui8>
    %3 = stablehlo.compare  NE, %2, %c : (tensor<3xui8>, tensor<3xui8>) -> tensor<3xi1>
    return %3 : tensor<3xi1>
  }
}
```
torch-xla converts logical boolean operations to ui8 ones. So, with the workaround, I got the model running. 

I also added eager mode. Now, I'm working on tensor parallel.
